src/Storefront/Resources/app/storefront/src/main.js

```diff
@@ -142,11 +143,26 @@ if (window.gtagActive) {
     PluginManager.register('GoogleAnalytics', () => import('src/plugin/google-analytics/google-analytics.plugin'));
 }
 
-if (window.googleReCaptchaV2Active) {
+// This logic ensures `grecaptcha.ready()` can be safely called at any time.
+// Callbacks passed to `grecaptcha.ready()` before reCAPTCHA loads are enqueued
+// and executed by the reCAPTCHA script once it loads.
+if ((window.googleReCaptchaV2Active || window.googleReCaptchaV3Active) && typeof window.grecaptcha === 'undefined') {
+    window.grecaptcha = {
+        ready: (cb) => {
+            const c = '___grecaptcha_cfg';
+            window[c] = window[c] || {};
+            window[c].fns = window[c].fns || [];
+            window[c].fns.push(cb);
+        },
+    };
+}
+
+const cookiesAccepted = CookieStorage.getItem('cookie-preference') === '1';
```

Comment:
Do we have a dedicated cookie preference/item for "technically required"? (which includes captcha?) Because this just checks if ANY cookie preference is set. Or can we assume that when cookie-preference is 1 it will always include technically required = true?

Reply:
A code comment would be nice then, so everyone understands the magic number. 😄